### PR TITLE
docs: drop bazel

### PR DESCRIPTION
**Pull request by tao12345666333**

### CODE REVIEW

File: site/content/docs/user/quick-start.md [hunk 1]
```diff
@@ -202,18 +202,15 @@ The `node-image` in turn is built off the [`base-image`][base image], which
...
-Note however that using `--type=bazel` on Windows or MacOS will not work
-currently due to Kubelet using [CGO] which requires GCC/glibc for linux.
-A workaround may be enabled in the future.
 
-kind will default to using the build type `docker` if none is specified.
+kind will default to using the build type `docker`.
 
 ```
-kind build node-image --type bazel
+kind build node-image --type docker
```
Comment (BenTheElder) [inline site/content/docs/user/quick-start.md:213]: this section and flag should be dropped entirely, the flag is deprecated.

Comment (tao12345666333) [reply site/content/docs/user/quick-start.md]: Addressed.

File: README.md [hunk 2]
```diff
@@ -115,7 +115,7 @@ See also: our own [contributor guide] and the Kubernetes [community page].
 
 - kind supports multi-node (including HA) clusters
 - kind supports building Kubernetes release builds from source
-  - support for make / bash / docker or bazel, in addition to pre-published builds
+  - support for make / bash or docker, in addition to pre-published builds
```
Comment (BenTheElder) [inline README.md:118]: this is the same thing currently, we should drop this bullet. can do that nit in a follow up as cleaning up bazel on it's own is worth it

File: pkg/cmd/kind/build/nodeimage/nodeimage.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func NewCommand(logger log.Logger, streams cmd.IOStreams) *cobra.Command {
 	}
 	cmd.Flags().StringVar(
 		&flags.BuildType, "type",
-		"docker", "build type, one of [bazel, docker]",
+		"docker", "build type, default is docker",
```
Comment (BenTheElder) [inline pkg/cmd/kind/build/nodeimage/nodeimage.go:54]: TODO: document deprecation in usage and/or hide flag

**Comment by BenTheElder** [comment]
/lgtm
/approve
will add more details to this in a follow-up as we need to put a note regarding that kind requires all of upstream kubernetes's build requirements and link to the upstream docs xref https://github.com/kubernetes-sigs/kind/issues/2188

**Comment by BenTheElder** [comment]
thank you!

**Pull request merged**